### PR TITLE
Visual update for menu flyout

### PR DESCRIPTION
**Pull request by tashatitova**
Most big "deletions" in this PR are not deletions - I moved legacy styles to the bottom of the file in order to clean up a bit without removing them (to avoid possible regressions if there are custom styles existing based off of the legacy styles).

Actual changes include:
- Moving non-theming resources to static section
- Updating the padding/margins accordingly: spacing between items, spacing for the separator (extending it edge to edge required negative margins)
- Adding rounded corners on menu items to show up in interaction states
- Updating brushes 
- Updating chevron sizes
- Updating the accelerator text margins to make the text base line the same for item text and accelerator text (both were vertically aligned center and text looked off at the bottom)
- Redirect default item style from Reveal to default

Note: not fixing the selected item alignment to the selected item in opened submenu. Followed the thread where the decision was to not change the flyout positioning, and the margins change here is not quite possible to work in all scenarios.

dark-icons
dark-1
dark-2

light-icons
light-1
light-2

hc-1
hc-2
hc-3

### CODE REVIEW

**Review by teaP — APPROVED** [review]

**Comment by duke7553** [comment]
Does this change imply that reveal isn't recommended anymore on MenuFlyout Items?

**Comment by mdtauk** [comment]
> Wow, looks great! 
> 
> Question:
> Does this change imply that reveal isn't recommended anymore on MenuFlyout Items?

As of right now, none of the reveal styles have been changed with these visual updates.

There was a discussion that @chigy posted some time back, where they are considering changes to the Reveal effect. #1126 

So I'm hoping they will update the Reveal Button Styles and other styles when they have it ready.

WinUI 3 won't support reveal at launch, but I assume these new visual styles will come to WinUI 2.6 and onwards

**Comment by chigy** [comment]
> > Wow, looks great!
> > Question:
> > Does this change imply that reveal isn't recommended anymore on MenuFlyout Items?
> 
> As of right now, none of the reveal styles have been changed with these visual updates.
> 
> There was a discussion that @chigy posted some time back, where they are considering changes to the Reveal effect. #1126
> 
> So I'm hoping they will update the Reveal Button Styles and other styles when they have it ready.
> 
> WinUI 3 won't support reveal at launch, but I assume these new visual styles will come to WinUI 2.6 and onwards

Glad you like it. :)

Yes, we are going away with reveal effect. With the new styles of visuals, reveal just does not work as an effective visual effect to express interaction. We are looking into micro-animations or other types effects (you might have seen in some other controls we are working on) to ensure controls get the interaction communicated. That said, I would say, consider this as we are halting on reveal. We may find a good use for it in the future (or not...). We just haven't found effective one at the moment...

**Comment by mdtauk** [comment]
@chigy  Will you be considering effects for the borders of controls, most of the controls have borders now with the visual refresh (except those using Ghost ThemeResources) - if not Reveal as it is, but some kind of visual or colour flare for hover approaches?

**Comment by chigy** [comment]
> @chigy Will you be considering effects for the borders of controls, most of the controls have borders now with the visual refresh (except those using Ghost ThemeResources) - if not Reveal as it is, but some kind of visual or colour flare for hover approaches?

@mdtauk , no not at this time. We try to be thoughtful on adding effects or visual feedback and be systematic about it. We are currently busy with the visual refresh this is not something we are focused on at the time.

**Comment by Shomnipotence** [comment]
Why does this projection look so close to the appearance of macOS...

**Comment by tashatitova** [comment]
> Wow, looks great!
> 
> Question:
> Does this change imply that reveal isn't recommended anymore on MenuFlyout Items?

Reveal isn't the default style anymore, that's correct.

**Comment by nikhil-anilkumar** [comment]
The rounded corners seem to have too much of a corner radius. Hopefully it reduces to 1px or 2px in the final implementation. A subtle rounded corner is much more comfortable for mouse pointer users, while also staying intuitive for touch users.

**Comment by SeanLedbetter** [comment]
Please don't put drop shadows or reveal effects. Artifacts are bad design. The corners are too rounded too. This looks awful. More of the same awful. The same gaudy effects. Also, a black border over dark grey is hideous like the "Keyboard accelerator" button. The border should be a lighter grey, since it's a **highlight**. Crap like this for eight years since Windows 8. Windows 8 with it's horizontal apps and off-screen chrome. I could go into the OneDrive app on Windows 10X and why it is a nightmare to look at. Since I brought it up, Windows 10X has a lot of problems just with the opening screen. You can't put anything on the desktop background, So it's just an empty background for a picture. The taskbar still resides on the edge of the screen, instead of hovering off it which would make it easier for touch targets. Also, that **brand new taskbar** is bad design as well. What is all of that space in the taskbar that doesn't have apps in it going to be used for anyway? Especially, since this has a Start Menu. Uck, and instead of the system tray icons lined up on the right, they are stacked. Uck. Between this and the app layout of apps like the new OneDrive app on Windows 10X.

[Here's a post on the Your Phone app.](https://www.reddit.com/r/Windows10/comments/gao80h/the_problem_with_providing_feedback_to_microsoft/)

**Comment by Shomnipotence** [comment]
image
Pull，The projection of the area where the menus are stacked looks too strange
image
macOS

**Comment by Shomnipotence** [comment]
Is it just me that I can hardly see the current highlighted item? Is this contrast insufficient for recognition?

**Comment by tsukiaru** [comment]
> Is it just me that I can hardly see the current highlighted item? Is this contrast insufficient for recognition?

Agreed. Perhaps it should use the system accent colour? Unless that's not possible.

File: dev/MenuFlyout/MenuFlyout_themeresources.xaml [hunk 1]
```diff
@@ -600,304 +449,459 @@
...
                                 FontFamily="{ThemeResource SymbolThemeFontFamily}"
                                 Glyph="&#xE001;"
-                                FontSize="16"
-                                Foreground="{ThemeResource ToggleMenuFlyoutItemCheckGlyphForeground}"
+                                FontSize="12"
+                                Foreground="{ThemeResource MenuFlyoutSubItemChevron}"
                                 Opacity="0"
-                                Width="16"
-                                Margin="0,0,12,0" />
-                            <Viewbox x:Name="IconRoot"
```
Comment (StephenLPeters) [inline dev/MenuFlyout/MenuFlyout_themeresources.xaml:729]: Looks like we removed the IconRoot from the toggle menu flyout items here. Seems like a mistake

Comment (tashatitova) [reply dev/MenuFlyout/MenuFlyout_themeresources.xaml:729]: @StephenLPeters definitely seems an accidental removal

**Pull request merged**